tests/test_optimization.py

```
@@ -0,0 +1,121 @@
+    # Set up a simple test graph
+    num_cyclins = len(calc._all_cyclins)
+    test_graph = jnp.ones((num_cyclins, num_cyclins))  # Simple all-connected graph
+    calc.set_custom_connected_graph(test_graph.tolist(), "TestGraph")
+
+    # Test state iteration
+    print(f"Number of Starting States: {len(calc._start_states)}")
+    print(f"Number of G1 States: {len(calc._g1_start_states)}")
+
+    # Run a small test
+    try:
+        graph_score, final_state_count, state_seq_types = calc.generate_graph_score_and_final_states()
```

Pylint (3.3.7)

[convention] Line too long (102/100)

(C0301)